File: Javascript/Largest-Perimeter-Triangle.js
/**
 * @param {number[]} nums
 * @return {number}
 */
const largestPerimeter = (nums) => {
  nums.sort((a, b) => b - a);

  let maxPerimeter = 0;

  for (let i = 0; i < nums.length - 2; i++) {
    if (nums[i] < nums[i + 1] + nums[i + 2]) {
      maxPerimeter = Math.max(
        maxPerimeter,
        nums[i] + nums[i + 1] + nums[i + 2]
      );
      break;
    }
  }

  return maxPerimeter;
};

File: Rust/Largest-Perimeter-Triangle.rs
pub struct Solution;

impl Solution {
    pub fn largest_perimeter(nums: Vec<i32>) -> i32 {
        let mut nums = nums;
        nums.sort_by(|a, b| b.partial_cmp(a).unwrap());

        let mut max_perimeter = 0;

        for (index, length) in nums.iter().enumerate() {
            if index < nums.len() - 2 && *length < nums[index + 1] + nums[index + 2] {
                max_perimeter = max_perimeter.max(*length + nums[index + 1] + nums[index + 2]);
                break;
            }
        }

        max_perimeter
    }
}

fn main() {}
